File: engine/engines_retrieve.py
from twelvelabs import TwelveLabs
import os

client = TwelveLabs(api_key=os.environ.get("API_KEY"))

retrieved_engine = client.engine.retrieve(id="pegasus1.1")
print(f"Author: {retrieved_engine.author}")
print(f"Allowed engine options: {retrieved_engine.allowed_engine_options}")
print(f"Ready: {retrieved_engine.ready}")
print(f"Finetune: {retrieved_engine.finetune}")

File: engine/engines-retrieve.js
import { TwelveLabs } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY}); 

  const retrievedEngine = await client.engine.retrieve('pegasus1.1')
  console.log(`Author: ${retrievedEngine.author}`);
  console.log(`Allowed engine options: ${retrievedEngine.allowedEngineOptions}`);
  console.log(`Ready: ${retrievedEngine.ready}`);
  console.log(`Finetune: ${retrievedEngine.finetune}`);
})();
